### PR TITLE
UPSTREAM: 33024: Make status errors with  multiple causes cleaner

### PR DESCRIPTION
**Pull request by juanvallejo**
UPSTREAM: https://github.com/kubernetes/kubernetes/pull/33024
Related BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1355703

Invalid status errors for resources are printed without any formatting.
This makes an error message hard to read when the error contains
multiple causes.

This patch loops through all of the error's causes, and prints each one
in its own line.

##### Before
```
$ oc set probe dc/hello --liveness --get-url="www.sample.com:80"

error: DeploymentConfig "hello" is invalid: [spec.template.spec.containers[0].livenessProbe.httpGet.port: 
Invalid value: "": must be an IANA_SVC_NAME (at most 15 characters, matching regex [a-z0-9]([a-z0-9-]*
[a-z0-9])*, it must contain at least one letter [a-z], and hyphens cannot be adjacent to other hyphens): e.g.
"http", spec.template.spec.containers[0].livenessProbe.httpGet.scheme: Invalid value: 
"WWW.SAMPLE.COM": must be one of [HTTP HTTPS]]
```

##### After

```
$ oc set probe dc/hello --liveness --get-url="www.sample.com:80"
error: DeploymentConfig "hello" is invalid

* spec.template.spec.containers[0].livenessProbe.httpGet.port: Invalid value: "": must be an IANA_SVC_NAME (at most 15 characters, matching regex [a-z0-9]([a-z0-9-]*[a-z0-9])*, it must contain at least one letter [a-z], and hyphens cannot be adjacent to other hyphens): e.g. "http"
* spec.template.spec.containers[0].livenessProbe.httpGet.scheme: Invalid value: "WWW.SAMPLE.COM": must be one of [HTTP HTTPS]
```

@openshift/cli-review

### CODE REVIEW

File: pkg/cmd/cli/cmd/set/helper.go [hunk 1]
```diff
@@ -42,6 +42,12 @@ func handlePodUpdateError(out io.Writer, err error, resource string) {
 			if all && match {
 				return
 			}
+		} else {
+			fmt.Fprintf(out, "error: %s %q is invalid\n\n", errorDetails.Kind, errorDetails.Name)
```
Comment (smarterclayton) [inline pkg/cmd/cli/cmd/set/helper.go]: We should be using the multi-error output from kcmdutil here (or putting a method in place there that does this consistently).  I.e. we want a "print err with causes" helper in kcmdutil.

Comment (juanvallejo) [inline pkg/cmd/cli/cmd/set/helper.go]: @smarterclayton Sounds good, PTAL: https://github.com/openshift/origin/pull/10965/commits/c5da4b2808cc6abb7f0e7bf469ce9d66916485b1

**Comment by juanvallejo** [comment]
[test]

**Comment by juanvallejo** [comment]
[conformance](https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_conformance/6352/) test flaked on https://github.com/openshift/origin/issues/9548

**Comment by juanvallejo** [comment]
[conformance](https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_conformance/6357/) test flaked on https://github.com/openshift/origin/issues/11008 re[test]

**Comment by juanvallejo** [comment]
[conformance](https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_conformance/6399/) test flaked on https://github.com/openshift/origin/issues/9548 re[test]

**Comment by juanvallejo** [comment]
@fabianofranz Got LGTM upstream! This PR should be up to date with upstream changes

**Comment by fabianofranz** [comment]
LGTM [merge]

**Pull request merged**